api/v1alpha1/regexp.go

```diff
@@ -0,0 +1,20 @@
+package v1alpha1
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// ValidateRegularExpression ensures that the provided regular expression
+// matches the expected results for the given test cases. Each key in the
+// testCases is tested against the expression, and the result must match the
+// value in the map of testcases.
+func ValidateRegularExpression(t *testing.T, expression *regexp.Regexp, testCases map[string]bool) {
```

Comment:
I get how the constraints are cornering us into doing this, but I would at least like to see this in `v1alpha1/testing` to clearly indicate it is not part of the regular API.

Reply:
I totally agree.  "testing" is unfortunate though, since it would conflict with the testing package that is imported in all tests.  I'll called it "testhelper"

Reply:
`testutils` would be more idiomatic I think, `testing` also really wouldn't be a problem as you can just alias it as `apitesting` during import in the couple of places where we use it.